gravitee-am-gateway/gravitee-am-gateway-handler/gravitee-am-gateway-handler-core/src/main/java/io/gravitee/am/gateway/handler/root/resources/handler/error/ErrorHandler.java

```diff
@@ -97,29 +99,41 @@ private void handleException(RoutingContext routingContext, String errorCode, St
             } else if (request.getParam(Parameters.CLIENT_ID) != null) {
                 parameters.set(Parameters.CLIENT_ID, (request.getParam(Parameters.CLIENT_ID)));
             }
+            StringBuilder error = new StringBuilder();
             // append error information
             parameters.set(ConstantKeys.ERROR_PARAM_KEY, errorCode);
+            error.append(errorCode);
             if (errorDetail != null) {
                 parameters.set(ConstantKeys.ERROR_DESCRIPTION_PARAM_KEY, errorDetail);
+                error.append("$");
+                error.append(errorDetail);
             }
 
+            String errorHash = HashUtil.generateSHA256(error.toString());
+
             final String token = request.getParam(TOKEN_CONTEXT_KEY);
             if (isNull(parameters.get(TOKEN_CONTEXT_KEY)) && token != null) {
                 parameters.set(TOKEN_CONTEXT_KEY, token);
             }
 
             // redirect
             String proxiedErrorPage = UriBuilderRequest.resolveProxyRequest(request, errorPageURL, parameters, true);
-            doRedirect(routingContext, proxiedErrorPage);
+            doRedirect(routingContext, proxiedErrorPage, errorHash);
         } catch (Exception e) {
             logger.error("Unable to handle root error response", e);
-            doRedirect(routingContext, errorPageURL);
+            doRedirect(routingContext, errorPageURL, null);
         }
     }
 
-    private void doRedirect(RoutingContext context, String url) {
-        if (destroySession && context.session() != null) {
-            context.session().destroy();
+    private void doRedirect(RoutingContext context, String url, String errors) {
+
+        if (context.session() != null) {
+            if (destroySession) {
+                context.session().destroy();
+            }
+            if (errors != null) {
+                context.session().put(ERROR_HASH, errors);
+            }
```

Comment:
If `destroySession` is `true`, and there's an error this code will destroy the session (clearing all the data), and then put the error hash in in (which essentially starts a new session, with a new session id; see `io.vertx.ext.web.sstoreio.vertx.ext.web.sstore.AbstractSession` `#put()`, `#data()`)

Is that the behaviour we want?

Reply:
Yes, we kind of cleaning up session and then recreate only with error hash.